### PR TITLE
Sync sandeep

### PR DESCRIPTION
**Pull request by Gwillison415**

### CODE REVIEW

**Comment by michaelmurray6298** [comment]
Hey grant, you're trying to merge this into master.

**Comment by Gwillison415** [comment]
Changed base from master to dev

**Comment by HamidAghdaee** [comment]
@SSabu @Gwillison415 this PR has merge conflicts. Let me know when it is actionable.

File: package.json [hunk 1]
```diff
@@ -4,16 +4,19 @@
...
+    "lodash": "^4.17.4",
     "prop-types": "15.5.10",
     "react": "15.6.1",
+    "react-cookies": "^0.1.0",
     "react-dom": "15.6.1",
     "react-linechart": "1.1.9",
     "react-redux": "5.0.5",
     "redux": "3.7.2",
-    "redux-promise-middleware": "4.3.0",
+    "redux-promise-middleware": "^4.3.0",
```
Comment (HamidAghdaee) [inline package.json]: please don't use ^

File: src/App.test.jsx [hunk 2]
```diff
@@ -8,6 +8,23 @@ import thunkMiddleware from 'redux-thunk';
...
+import configureStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import Api from './utils/Api';
+import thunkMiddleware from 'redux-thunk';
+import { Provider } from 'react-redux';
+import rootReducer from './reducers/index';
+
+const middleware = [
+  promiseMiddleware(),
+  thunkMiddleware.withExtraArgument({ Api}),
```
Comment (HamidAghdaee) [inline src/App.test.jsx]: why both Thunk and thunkmiddleware.withExtraargument?

Comment (HamidAghdaee) [inline src/App.test.jsx]: Actually, why not just shallow render the unconnected app component? that way you won't need any of the redux stuff

File: src/actions/renderActions.js [hunk 3]
```diff
@@ -1,16 +1,15 @@
...
 export const RENDER_REPOS = 'RENDER_REPOS';
 export const RENDER_DASHBOARD = 'RENDER_DASHBOARD';
+export const GET_USERNAME = 'GET_USERNAME';
+// const AUTH_ROUTE = 'http://localhost:8000/isAuthenticated';
 
-export const renderOrgs = userName => (dispatch) => {
-  dispatch({
-    type: RENDER_ORGS,
-    responseStr: 'orgs',
+export const renderOrgs = (userName) => (dispatch) => {
```
Comment (HamidAghdaee) [inline src/actions/renderActions.js]: I don't think this is a good name for this action creater, it actually doesn't render anything (which it shouldn't, that's not the job of action creators)

File: src/store.js [hunk 4]
```diff
@@ -21,10 +20,9 @@ const logger = store => next => (action) => {
 // };
 
 const middleware = [
-  // thunk,
-  thunkMiddleware.withExtraArgument({ Api }),
-  logger,
-  // error,
+  promiseMiddleware(),
```
Comment (HamidAghdaee) [inline src/store.js]: why both thunk and thunkMiddidlewarewithextraarg?

File: src/test/__snapshots__/PrTable.test.jsx.snap [hunk 5]
```diff
@@ -9,5 +9,11 @@ exports[`PrTable component should render a component with props as specified  1`
       Pull Requests
     </CardHeader>
   </CardContent>
+<<<<<<< HEAD
```
Comment (HamidAghdaee) [inline src/test/__snapshots__/PrTable.test.jsx.snap]: a bunch of the test files seem to have merge conflict stuff in them, like this.

**Comment by HamidAghdaee** [comment]
@Gwillison415 see my comments

File: src/App.test.jsx (hunk 2 above)
Comment (Gwillison415) [reply src/App.test.jsx]: because: multiple developers.. @SSabu what do you think we should go with?

**Comment by HamidAghdaee** [comment]
@Gwillison415 @SSabu So many merge conflicts here.

**Comment by HamidAghdaee** [comment]
we agreed that @Gwillison415 is going to withdraw this PR and commit code for the OAuth stuff on a new branch built off upstream dev.

**Pull request closed without merging**